### DISCUSSION TITLE
Redirect trouble after login

### DISCUSSION THREAD

**Discussion by fbryo21** [nextauthjs/next-auth#7456] [Help]
### Question 💬

Can anyone help me? I'm on trouble, settings on redirecting back to protected page

I reproduce it on StackOverflow:
https://stackoverflow.com/q/76153631/12574063

### How to reproduce ☕️

export default NextAuth({
    adapter: MongoDBAdapter(clientPromise),
    secret: process.env.NEXTAUTH_SECRET,
    providers: [
        // OAuth authentication providers
        GoogleProvider({
            clientId: process.env.GOOGLE_ID,
            clientSecret: process.env.GOOGLE_SECRET,
        }),
    ],
    callbacks: {
        async redirect({ url, baseUrl }) {
            // Allows relative callback URLs
            if (url.startsWith("/")) return `${baseUrl}${url}`
            // Allows callback URLs on the same origin
            else if (new URL(url).origin === baseUrl) return url
            return baseUrl
        }
}
})

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by JasperAlexander** [comment]
You can set a redirect URI in Google.
